### PR TITLE
DUPLO-8877 external_port is now optional, sometimes

### PR DESCRIPTION
**Pull request by ddiaz-duplo**
## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

### Input
```
resource "duplocloud_duplo_service_lbconfigs" "myservice2" {
  tenant_id                   = duplocloud_duplo_service.myservice2.tenant_id
  replication_controller_name = duplocloud_duplo_service.myservice2.name

  lbconfigs {
    health_check_url = "/"
    is_native        = false
    lb_type          = 5 # Application load balancer
    port             = "80"
    protocol         = "http"
  }
}

```
### Output
```
duplocloud_duplo_service_lbconfigs.myservice2: Creating...

│ Error: 'external_port' is required when 'lb_type' is set to 5
│ 
│   with duplocloud_duplo_service_lbconfigs.myservice2,
│   on main.tf line 99, in resource "duplocloud_duplo_service_lbconfigs" "myservice2":
│   99: resource "duplocloud_duplo_service_lbconfigs" "myservice2" {
```

### Development

- [x] Pull requests may not be submitted to the *master* branch (use *develop* instead) - or they will be closed.
- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-8877 TF: External port should not be required field for Target group only lb type](https://app.clickup.com/t/8655600/DUPLO-8877)

File: duplosdk/replication_controller.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ type DuploLbConfiguration struct {
 	Protocol                  string                    `json:"Protocol"`
 	Port                      string                    `json:"Port"`
 	HostPort                  int                       `json:"HostPort"`
-	ExternalPort              int                       `json:"ExternalPort"`
+	ExternalPort              *int                      `json:"ExternalPort,omitempty"`
```
[the remark this answers is not in the document]
Comment (joek-duplo) [reply duplosdk/replication_controller.go:78]: the backend will actually have it as `0`, right?  it's just an `int` on the backend isn't it?   we should match the backend - even though it's lame.

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: duplosdk/replication_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddiaz-duplo) [reply duplosdk/replication_controller.go:78]: The backend will not set it to 0 if you omit it from the JSON. It sets it to match the `Port`. 

# Testing
Run this curl command to see result. Note that I set `Port` to 81

## Call API
```bash
curl --location 'https://test19.duplocloud.net/subscriptions/81386040-9a07-4e76-a195-94191d9d47aa/LBconfigurationBulkUpdate' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer xxxxxxxx' \
--data '{
    "TenantId":"81386040-9a07-4e76-a195-94191d9d47aa",
    "Name":"myservice",
    "LBConfigurations":
    [
        {
            "TenantId":"81386040-9a07-4e76-a195-94191d9d47aa",
            "ReplicationControllerName":"myservice",
            "LbType":7,
            "Protocol":"http",
            "Port":"81",
            "HostPort":0,
            "HealthCheckUrl":"/",
            "LbIndex":0,
            "ExtraSelectorLabels":null
        }
    ]
}'
```

## Results

image

**Review by joek-duplo — APPROVED** [review]
nice work, and you proved me to wrong to prevent a regression I would have caused.  thanks!

**Pull request merged**